examples/self-diagnostics/src/main.rs

```diff
@@ -72,11 +74,22 @@ fn init_logger_provider() -> opentelemetry_sdk::logs::LoggerProvider {
         .add_directive("hyper=error".parse().unwrap())
         .add_directive("tonic=error".parse().unwrap())
         .add_directive("reqwest=error".parse().unwrap());
-    let cloned_provider = provider.clone();
-    let layer = layer::OpenTelemetryTracingBridge::new(&cloned_provider);
+    let opentelemetry_filter =
+        tracing_subscriber::filter::filter_fn(|metadata| metadata.target() == "opentelemetry");
+
+    let non_opentelemetry_filter =
```

Comment:
I believe we already have decided in a previous SIG meeting to refactor this example. Can you clean it up.
IMHO, all it needs to show is
1. How to setup gathering internal logs via a layer
2. How to avoid sending the internal logs back to otel (show the filter).
3. Show the common case of hyper,tonic etc filtering.

Reply:
Can you clarify what to cleanup, as I believe the current example is just doing above three.

Reply:
The part where the examples show storing 1st message and ignoring rest.
![image](https://github.com/user-attachments/assets/e96f03f1-da61-4d87-97ca-a24faf3d2b51)